src/main/routes/payment-plan-calculation.ts

```diff
@@ -0,0 +1,47 @@
+import * as express from 'express'
+import * as HttpStatus from 'http-status-codes'
+import * as _ from 'lodash'
+
+import { Paths as AppPaths } from 'paths'
+import { createPaymentPlan } from 'common/paymentPlan'
+
+/* tslint:disable:no-default-export */
+export default express.Router()
+  .get(AppPaths.paymentPlanCalculation.uri, (req, res) => {
+
+    const totalAmount = req.query['total-amount']
+    const instalmentAmount = req.query['instalment-amount']
+    const frequencyInWeeks = req.query['frequency-in-weeks']
+
+    const error = validate(totalAmount, instalmentAmount, frequencyInWeeks)
+
+    if (error) {
+      return res.status(HttpStatus.UNPROCESSABLE_ENTITY).json({
+        error: {
+          status: HttpStatus.UNPROCESSABLE_ENTITY,
+          message: error
+        }
+      })
+    }
+
+    const paymentPlan = createPaymentPlan(totalAmount, instalmentAmount, frequencyInWeeks)
+
+    return res.status(HttpStatus.OK).json({
+      paymentPlan: {
+        paymentLength: paymentPlan.getPaymentLength(),
+        lastPaymentDate: paymentPlan.getLastPaymentDate().toJSON()
+      }
+    })
+  })
+
+function validate (totalAmount, instalmentAmount, frequencyInWeeks) {
+  if (_.isEmpty(totalAmount)) {
```

Comment:
I would love to add extra validation to make sure that:
- values are of numeric type
- values are positive numbers

Reply:
Done